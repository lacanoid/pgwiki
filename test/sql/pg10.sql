\pset null _null_
\pset format unaligned

SET client_min_messages = warning;
SET ROLE postgres;
------
CREATE TABLE options (
       name text primary key,
       value text not null,
       regtype regtype not null default('text'::regtype)
);
ALTER TABLE options SET ( parallel_workers = 2 );
select ddlx_script('options');
------
CREATE TABLE measurement (
    city_id         int not null,
    logdate         date not null,
    peaktemp        int,
    unitsales       int
) PARTITION BY RANGE (logdate);

CREATE TABLE measurement_y2006m02 PARTITION OF measurement
    FOR VALUES FROM ('2006-02-01') TO ('2006-03-01');

CREATE TABLE measurement_y2006m03 PARTITION OF measurement
    FOR VALUES FROM ('2006-03-01') TO ('2006-04-01');

CREATE INDEX ON measurement_y2006m02 (logdate);
CREATE INDEX ON measurement_y2006m03 (logdate);

select ddlx_script('measurement');
------
CREATE TABLE customers(cust_id bigint NOT NULL,cust_name varchar(32) NOT NULL,cust_address text,
cust_country text) PARTITION BY LIST(cust_country);
CREATE TABLE customer_ind PARTITION OF customers FOR VALUES IN ('ind');
CREATE TABLE customer_jap PARTITION OF customers FOR VALUES IN ('jap');
INSERT INTO customers VALUES (2039,'Puja','Hyderabad','ind');
SELECT tableoid::regclass,* FROM customers;
SELECT * FROM customer_ind;
UPDATE customers SET cust_country ='jap' WHERE cust_id=2039;
SELECT * FROM customer_jap;

select ddlx_script('customers');
select ddlx_script('customer_jap'); 

create table log (
 ts timestamp,
 code int,
 t text,
 a boolean,
 d varchar collate "C",
 j json
)
partition by range (code,date_trunc('month',ts),t collate "C",a,((code%100)/100));
select ddlx_script('log');

-- statistics
CREATE TABLE test_stat (
    a   int primary key,
    b   int
);
CREATE STATISTICS test_stat1 (dependencies) ON a, b FROM test_stat;

\x
select ddlx_create(oid),ddlx_drop(oid) from pg_statistic_ext where stxname='test_stat1';
\x
select ddlx_script('test_stat');
