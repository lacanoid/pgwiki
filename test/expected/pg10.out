\pset null _null_
\pset format unaligned
SET client_min_messages = warning;
SET ROLE postgres;
------
CREATE TABLE options (
       name text primary key,
       value text not null,
       regtype regtype not null default('text'::regtype)
);
ALTER TABLE options SET ( parallel_workers = 2 );
select ddlx_script('options');
ERROR:  function ddlx_script(unknown) does not exist
LINE 1: select ddlx_script('options');
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
------
CREATE TABLE measurement (
    city_id         int not null,
    logdate         date not null,
    peaktemp        int,
    unitsales       int
) PARTITION BY RANGE (logdate);
CREATE TABLE measurement_y2006m02 PARTITION OF measurement
    FOR VALUES FROM ('2006-02-01') TO ('2006-03-01');
CREATE TABLE measurement_y2006m03 PARTITION OF measurement
    FOR VALUES FROM ('2006-03-01') TO ('2006-04-01');
CREATE INDEX ON measurement_y2006m02 (logdate);
CREATE INDEX ON measurement_y2006m03 (logdate);
select ddlx_script('measurement');
ERROR:  function ddlx_script(unknown) does not exist
LINE 1: select ddlx_script('measurement');
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
------
CREATE TABLE customers(cust_id bigint NOT NULL,cust_name varchar(32) NOT NULL,cust_address text,
cust_country text) PARTITION BY LIST(cust_country);
CREATE TABLE customer_ind PARTITION OF customers FOR VALUES IN ('ind');
CREATE TABLE customer_jap PARTITION OF customers FOR VALUES IN ('jap');
INSERT INTO customers VALUES (2039,'Puja','Hyderabad','ind');
SELECT tableoid::regclass,* FROM customers;
tableoid|cust_id|cust_name|cust_address|cust_country
customer_ind|2039|Puja|Hyderabad|ind
(1 row)
SELECT * FROM customer_ind;
cust_id|cust_name|cust_address|cust_country
2039|Puja|Hyderabad|ind
(1 row)
UPDATE customers SET cust_country ='jap' WHERE cust_id=2039;
ERROR:  new row for relation "customer_ind" violates partition constraint
DETAIL:  Failing row contains (2039, Puja, Hyderabad, jap).
SELECT * FROM customer_jap;
cust_id|cust_name|cust_address|cust_country
(0 rows)
select ddlx_script('customers');
ERROR:  function ddlx_script(unknown) does not exist
LINE 1: select ddlx_script('customers');
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
select ddlx_script('customer_jap'); 
ERROR:  function ddlx_script(unknown) does not exist
LINE 1: select ddlx_script('customer_jap');
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
create table log (
 ts timestamp,
 code int,
 t text,
 a boolean,
 d varchar collate "C",
 j json
)
partition by range (code,date_trunc('month',ts),t collate "C",a,((code%100)/100));
select ddlx_script('log');
ERROR:  function ddlx_script(unknown) does not exist
LINE 1: select ddlx_script('log');
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
-- statistics
CREATE TABLE test_stat (
    a   int primary key,
    b   int
);
CREATE STATISTICS test_stat1 (dependencies) ON a, b FROM test_stat;
\x
select ddlx_create(oid),ddlx_drop(oid) from pg_statistic_ext where stxname='test_stat1';
ERROR:  function ddlx_create(oid) does not exist
LINE 1: select ddlx_create(oid),ddlx_drop(oid) from pg_statistic_ext...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
\x
select ddlx_script('test_stat');
ERROR:  function ddlx_script(unknown) does not exist
LINE 1: select ddlx_script('test_stat');
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
